src/NetscriptWorker.ts

```diff
@@ -88,11 +88,16 @@ function startNetscript2Script(player: IPlayer, workerScript: WorkerScript): Pro
         "Concurrent calls to Netscript functions are not allowed! " +
         "Did you forget to await hack(), grow(), or some other " +
         "promise-returning function? (Currently running: %s tried to run: %s)";
-      if (runningFn) {
+      if (runningFn && !(runningFn === "sleep" && workerScript.exiting)) {
```

Comment:
Why ignore `sleep`? If we're 'exiting', shouldn't whatever previously running task already be canceled?

Reply:
I changed it (haven't worked on this PR in days now), but the idea was that if we are currently running `sleep` and we're exiting, then calling another function is allowed